web/src/connectors/edit-model-connector/edit-model-connector.tsx

```diff
@@ -0,0 +1,195 @@
+                                    basements={basements?.data}
+                                    models={models?.data}
+                                />
+                            </div>
+                            <div className={wizardStyles['content__footer']}>
+                                {renderWizardButtons({
+                                    onNextClick: handleNext
+                                })}
+                            </div>
+                        </>
+                    )
+                },
+                {
+                    title: 'Data',
+                    content: (
+                        <>
+                            <div className={wizardStyles['content__body']}>
+                                <EditModelData lens={lens} />
+                            </div>
+                            <div className={wizardStyles['content__footer']}>
+                                {renderWizardButtons({
+                                    onPreviousClick: handlePrev,
+                                    onNextClick: () => {
+                                        save();
+                                    },
+                                    nextButtonCaption: finishButtonCaption
+                                })}
+                            </div>
+                        </>
+                    )
+                }
+            ];
+            return <Wizard steps={steps} returnUrl={MODELS_PAGE} stepIndex={stepIndex} />;
+        },
+        [stepIndex, basements]
+    );
+
+    const editModelMutation = useEditModelMutation();
+
+    const handleOnModelEdit = useCallback(async (values: ModelValues) => {
+        let categories: (string | number)[] = [];
+        if (values.jobs) {
+            values.jobs.forEach((el) => {
+                categories = [...categories, ...el.categories];
+            });
+        } else {
+            values.categories?.forEach((el) => {
+                categories = [...categories, el.name];
+            });
+        }
+        categories.map((el) => {
+            return el.toString();
+        });
+        const model: Model = {
+            id: values.id,
+            name: values.name || '',
+            basement: values.basement?.id || '',
+            categories: categories as string[],
+            type: values.type,
+            training_id: values.training_id,
+            score: parseInt(values.score || ''),
+            data_path: {
+                file: values.data_path_file || '',
+                bucket: values.data_path_bucket || ''
+            },
+            configuration_path: {
+                file: values.configuration_path_file || '',
+                bucket: values.configuration_path_bucket || ''
+            }
+        };
+
+        console.log(model);
+
+        try {
+            const response = await editModelMutation.mutateAsync(model);
+            values.id = response.id;
```

Comment:
Do we really need to mutate `values` here?

Reply:
I suppose yes, for now, until the models are refactored